### DISCUSSION TITLE
Reporting functionality

### DISCUSSION THREAD

**Discussion by rwahyudi** [dundee/gdu#59] [Ideas]
I love the software - works very well on NAS with millions of files! 
parallelism improves performance because NAS is backed with hundreds of disks but network latency is killing the performance, 

- cold cache 
- 15TB NFS  - 3.2Million files 
- Connection : 10Gb . Client and NAS server are on the same layer 2 network 

```
time ncdu -1xo- /nfs > /tmp/ncdu.json
real    12m26.976s
user    0m17.116s
sys     0m46.036s

sync; echo 3 | sudo tee /proc/sys/vm/drop_caches
time gdu /nfs > /tmp/report.txt
real    1m36.599s
user    0m15.076s
sys     0m53.848s
```

Feature request which I think useful for a lot of people is to have the ability to save the report into file. 
ncdu has -o switch to generate report file and you can instruct ncdu to read the report instead of scanning directories. 

The report generated could be useful for gdu-web and can be used to generate historical usage report.

**Comment by dundee** [comment]
Could you describe a bit more the scenario? Why would you create the report on the NAS machine and then analyse it on other machine when you can run the gdu in interactive mode on the NAS machine directly?

**Reply by rwahyudi** [reply]
Most enterprise NAS won't allow users to install software or allow SSH access. To be able to run ncdu/gdu , we have to export shares on the NAS, mount the share on a client host, then run ncdu/gdu on the client.

In most cases, NAS server is sandwiched between the client and the raw storage devices ( ie the disks) making the network latency as the bottleneck.  

![GitHub Logo](https://www.thinkmate.com/tm/images/article/iSCSI_SAN.gif)

Currently, I am in the middle of migrating our NAS appliance. Having a report of all directories in the filesystem would be useful because we can use the list of directories to run parallel rsync / robocopy. In our environment running single thread find / ncdu on the whole filesystem takes around a week to complete.

**Reply by dundee** [reply]
Makes sence. I will try to implement the same format `ncdu` uses.

**Reply by dundee** [reply]
Just released https://github.com/dundee/gdu/releases/tag/v5.3.0
